core/bin/zksync_server/src/main.rs

```diff
@@ -150,6 +125,40 @@ async fn main() -> anyhow::Result<()> {
         object_store_config: ObjectStoreConfig::from_env().ok(),
     };
 
+    let postgres_config = configs.postgres_config.clone().context("PostgresConfig")?;
+
+    if opt.genesis || is_genesis_needed(&postgres_config).await {
+        let network = NetworkConfig::from_env().context("NetworkConfig")?;
```

Comment:
Again: both here and in `zksync_core/src/lib.rs` a lot of code was ugly and is ugly.
I have opted out of refactoring it, since it's going to be removed in subsequent PRs.